uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/util/ColumnIndexUtilities.java

```diff
@@ -81,6 +81,9 @@ public static int getHeaderBlockEndColumnIndex(final List<GridColumn<?>> allColu
         if (candidateHeaderColumnIndex == allColumns.size() - 1) {
             return candidateHeaderColumnIndex;
         }
+        if (candidateHeaderColumnIndex >= allColumns.size()) {
```

Comment:
Yes, could you please explain a little bit. Just we are sure we do not cause index out of bounds exception somewhere else.

Reply:
When you double-click the last cell in a header and the first cell is hidden (scrolled).
This didn't happened before because the last header cell was not "clickable" when we had the "Description" column.

Reply:
So the issue is that the `headerColumnIndex` is wrong?

Where is the `getHeaderBlockEndColumnIndex(..)` method called from? How was the parameter value derived?

Reply:
Is called here: https://github.com/kiegroup/appformer/blob/331f358198533922cf9da61ec52bb6826b97fd19/uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/util/CellContextUtilities.java#L98

In my opinion this fix is correct, because `ColumnIndexUtilities` already "fixes" the `candidateHeaderColumnIndex` to a valid value in other situations.

Reply:
I'm not convinced it is the correct fix; however I was unable to debug in GWT (wozza, is it just me or has GWT development taken a performance turn for the worse recently!?!?!?) I still believe `candidateHeaderColumnIndex` should not exceed the number of columns in the grid. However this PR fixes the issue and therefore so as not to stand in the way of progress I'll approve....

Reply:
For me the performance is the same. :D

I still think the fix is the correct, because if you trace back the `candidateHeaderColumnIndex` comes from `ci.getUiColumnIndex()`. I think the `uiColumnIndex()` is right and the logic is good since `ColumnIndexUtilities.getHeaderBlockEndColumnIndex()` in my understands have the job of get the logical end block column index for that specific UI column.

Reply:
Sure, but `ci.getUiColumnIndex()` should NEVER be greater than the `gridModel.getColumns().size() - 1` otherwise something has gone wrong somewhere else...